crates/turbo-tasks-memory/src/task.rs

```diff
@@ -1178,58 +1575,101 @@ impl Task {
         func(&mut list[i])
     }
 
+    /// Access to a cell.
+    pub(crate) fn with_cell_mut_if_available<T>(
+        &self,
+        index: CellId,
+        func: impl FnOnce(&mut Cell) -> T,
+    ) -> Option<T> {
+        if let TaskMetaStateWriteGuard::Full(mut state) = self.state_mut() {
+            if let Some(list) = state.cells.get_mut(&index.type_id) {
+                if let Some(cell) = list.get_mut(index.index as usize) {
+                    return Some(func(cell));
+                }
+            }
+        }
```

Comment:
Nit: this could be a map chain: `self.state_mut().as_full().map(|state| ...).map(|list| ...).map(func)`